python/foglamp/services/core/server.py

```diff
@@ -314,6 +342,9 @@ def get_certificates(cls):
 
             try:
                 cls.is_rest_server_http_enabled = False if config['enableHttp']['value'] == 'false' else True
+                # https is a must if authentication method is certificate
+                if cls.auth_method == 'certificate':
+                    cls.is_rest_server_http_enabled = False
```

Comment:
why this is a must?

Reply:
Because certificate cannot be allowed to be uploaded via insecure channel even if it is a public key.